stock_request_non_internal/models/stock_request_abstract.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2017 Eficent Business and IT Consulting Services, S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models
+
+
+class StockRequest(models.AbstractModel):
+    _inherit = "stock.request.abstract"
+
+    warehouse_id = fields.Many2one(required=False)
+    location_id = fields.Many2one(domain=[])
```

Comment:
I suppose that the domain in stock_request does not match your module. Why not changing to a lambda method (in stock_request) to compute it and override it here ?

Reply:
Good idea :+1: